SQL-763: Update release.md to remove Git release step (#95)
mongodb/mongo-jdbc-driver: RELEASE.md

Wait for the evergreen version to finish, and ensure that the release task completes successfully.

#### Verify release artifacts
Check that the version just released is available in the [Sonatype Nexus Repo Manager](https://oss.sonatype.org/#nexus-search;quick~mongodb-jdbc).  
It is also a good idea to download the released artifacts mongodb-jdbc-x.x.x.jar and mongodb-jdbc-x.x.x-all.jar and verify that they are correctly 

> loading and the simple operations connecting, retrieving metadata and executing a query are all working.

Author: I created [SQL-791](https://jira.mongodb.org/browse/SQL-791) for this yesterday, because I also think we should automate this :) But in the meantime I thought it's good to have it here, as a safety net until we actually had the artifacts validation in Evergreen.

> loading and the simple operations connecting, retrieving metadata and executing a query are all working.

Reviewer: Sounds good -- could you add a comment to SQL-791 indicating that we should remove this line from the instructions as part of that ticket?

#### Verify release artifacts

> Check that the version just released is available in the [Sonatype Nexus Repo Manager](https://oss.sonatype.org/#nexus-search;quick~mongodb-jdbc)

Reviewer 2: ```suggestion
Check that the version just released is available in the [Sonatype Nexus Repo Manager](https://oss.sonatype.org/#nexus-search;quick~mongodb-jdbc).  
```
Adding spaces so the lines don't overrun

It is also a good idea to download the released artifacts mongodb-jdbc-x.x.x.jar and mongodb-jdbc-x.x.x-all.jar and verify that they are correctly 
loading and the simple operations connecting, retrieving metadata and executing a query are all working.

> It should appear to [Maven Central](https://search.maven.org/search?q=g:org.mongodb%20AND%20a:mongodb-jdbc) after a while.

Reviewer 2: ```suggestion
The release artifacts should appear on [Maven Central](https://search.maven.org/search?q=g:org.mongodb%20AND%20a:mongodb-jdbc) after a while.
```
